### ISSUE TITLE
fluid memory access / never read

### ISSUE THREAD

**Issue by Albrecht-S** [fltk/fltk#660]
Git current master, commit 4c8927c08344f4275941c4367377f6ca60019648.

I found questionable code in fluid by using clang's static code analyzer (scan-build). A preliminary suggested patch:
```diff
diff --git a/fluid/Fl_Type.cxx b/fluid/Fl_Type.cxx
index 5bf28e016..9fdd54b2a 100644
--- a/fluid/Fl_Type.cxx
+++ b/fluid/Fl_Type.cxx
@@ -306,7 +306,7 @@ Fl_Type *Fl_Type::prev_sibling() {
 Fl_Type *Fl_Type::next_sibling() {
   Fl_Type *n;
   for (n = next; n && n->level > level; n = n->next) ;
-  if (n->level==level)
+  if (n && n->level==level)
     return n;
   return 0;
 }
@@ -410,7 +410,7 @@ void Fl_Type::add(Fl_Type *p, Strategy strategy) {
     // we have current, t is the new node, p is the parent
     // find the next child of the parent after current
     //t->add(p); // add as a last child
-    Fl_Type *cc = current;
+    Fl_Type *cc = 0; // current; // This value is never read
     for (cc = current->next; cc; cc=cc->next) {
       if (cc->level<=this->level)
         break;
```

Both parts of the patch have issues:

1. I assume that the condition in the first hunk should read `n->level < level;` (note: replace '>' with '<'). Maybe the wrong comparison was copied from the method `prev_sibling()` above it? Please check and fix if I'm correct.

2. I commented the patch in part 2 with "`// current; // This value is never read`" which was the message given by the static code analyzer. Indeed this is so, but maybe the assignment was there for a reason? Please feel free to use the patch and remove the comment.

**Comment by MatthiasWM** [comment]
The first one is a bug, fixed in bd5a42eba8d0f6da3e98d9d8a294556577180c7c . The second one is meh, but whatever.

The `n->level > level` is correct. The code skips over possible children in the list. `prev_sibling()` had a bug though, returning the parent instead of NULL if it was the first sibling.

**Comment by Albrecht-S** [comment]
> The second one is meh, but whatever.

It's a warning. Period. There can be several reasons why you make an assignment that is "never read", for instance
- assignment to the wrong variable. This could be uncovered and then corrected. You fixed one **similar** error I reported recently, but that was an `unused variable` warning. There the assignment was to the wrong variable, instead of the unused variable.
- duplicated code, for whatever reason

... or it is just confusing for the reader. ;-)
